src/map/SemanticMarkers.tsx

```diff
@@ -88,3 +149,24 @@ function boundsToGeoBBox(bounds : LatLngBounds) : GeoBBox {
 	   northEast: [north, east] }
 }
 
+function updateMarkers(toChangeMarkers: Array<ReactElement<MarkerProps>>,
+                       sourceMarkers: Array<ReactElement<MarkerProps>>,
+                       hashDif: number) {
+  return toChangeMarkers.map((markerObj) => {
+    // Calculate the matching geohash
+    const sourceKey = markerObj.key as string;
+    const sourceHash = sourceKey.slice(0, -hashDif);
+
+    // Find the object with the matching geohash
+    const matchingMarkers = sourceMarkers.filter(obj => {
+      return obj.key === sourceHash
+    });
+
+    // Clone marker element with new position
+    const markerCloneProps = {
+      position: matchingMarkers[0].props.position
```

Comment:
Can matchingMarkers be empty here?  
And matchingMarkers[0] be undefined?

(In bizarre circumstances, of course.)

Reply:
Maybe in a bizarre circumstance but I think for that to be the case then the geohashes have to be bad. 

This only gets called when there are current and previous markers, and the current and previous hashes are different lengths. So a failure would occur only if we didn't find a matching has in each.

I could put in some error handling just in case.

Reply:
There **might** (just a hunch) be literally an edge case where you zoom out (I guess) and pan at the same time and for some reason there's no matching marker.

I think it's worth handling the potential error.

Reply:
Let me know if that change works for ya :D

Reply:
That's great, thanks.  Will merge today and do some more git admin!